docs/design-docs/arm/arm.md

```diff
@@ -0,0 +1,86 @@
+# Epiphany Platform backup design document
+
+Affected version: 0.10.x/0.11.x
+
+## Goals
+
+This document outlines an aproach to add (partial) ARM support to Epicli. Not finalized requirements so far:
+
+- ARMv8/ARM64 architecture
+- Centos 7 as operating system
+- "any" provider as we do not want to provide ARM infrastructure on any cloud providers jet through Terraform
+- Epiphany components: K8s, Kafka, ELK, Grafana, Prometheus, Kibana, HAProxy, Keycloak, PostgreSQL
+
+## Approach
+
+The 2 high level approaches that have been opted so far:
+
+1. Add “architecture” flag when using Epicli
+2. Add new OS (CentosARM64 fe.)
+
+Have 2 big disadvanges from the start:
+
+1. Will require an additional input which makes things more confusing as they will need supply not only the OS but also Architecture for (offline) install. This should not be needed as we can detect the architecture we are working on, on all required levels.
+2. Does not require additional input but this will lead to code duplication in the ```requirements``` role as we need to maintain ```download-requirements.sh``` for each OS and architecture then.
+
+That is why I opt for an approach where we don`t add any architecture flag or new additional OS. The architecture we can handle on the code level and on the OS level only the ```requirements.txt``` might be different for each as indicated by initial research [here.](./centos-arm-analysis.md).
+
+## Changes required
+
+### Repostitory role
+
+In the repository role we need to change the download of the requirements to support additional architectures as download requirements might be different as:
+
+- Some components/roles might not have packages/binaries/containers that support ARM
+- Some filenames for binaries will be different per architecture
+
+Hense we should make a requirements.txt for each architecture we want to support, for example:
+
+- requirements_x86_64.txt (Should be the default and present)
+- requirements_aarch64.txt
+
+The ```download-requirements.sh``` script should be able to figure out which one to select based on the output of:

```

Comment:
What if the server where we will be downloading packages will have different architecture than targeted one?

Reply:
We are not going to support combinations of architectures in a cluster. The same that currently all machines in a cluster need to run the same OS.